src/compiler/emitter.ts

```diff
@@ -147,6 +147,9 @@ namespace ts {
                     emittedFilesList.push(bundleInfoPath);
                 }
             }
+            else if (emittedFilesList && compilerOptions.emitDeclarationOnly && declarationFilePath) {

```

Comment:
I don't this this is right fix. We should set emitSkipped correctly instead (because emitDeclarationOnly does not skip emit by emitting only .d.ts)